src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchService.kt

```diff
@@ -81,6 +89,69 @@ class MatchService(
     }
   }
 
+  private fun nameMatch(matchRequest: MatchRequest): BoolQueryBuilder? {
+    with(matchRequest) {
+      return QueryBuilders.boolQuery()
+          .mustWhenTrue({ activeSentence }, "currentDisposal", "1")
+          .must(QueryBuilders.boolQuery()
+              .should(QueryBuilders.boolQuery()
+                  .mustWhenPresent("surname", surname)
+                  .mustWhenPresent("firstName", firstName)
+                  .mustWhenPresent("dateOfBirth", dateOfBirth)
+              )
+              .should(QueryBuilders.boolQuery()
+                  .mustWhenPresent("offenderAliases.surname", surname)
+                  .mustWhenPresent("offenderAliases.firstName", firstName)
+                  .mustWhenPresent("offenderAliases.dateOfBirth", dateOfBirth)
+              )
+          )
+    }
+  }
+
+  private fun partialNameMatch(matchRequest: MatchRequest): BoolQueryBuilder? {
+    with(matchRequest) {
+      return QueryBuilders.boolQuery()
+          .mustWhenTrue({ activeSentence }, "currentDisposal", "1")
+          .mustWhenPresent("surname", surname)
+          .mustWhenPresent("dateOfBirth", dateOfBirth)
+    }
+  }
+
+
+  private fun partialNameMatchDateOfBirthLenient(matchRequest: MatchRequest): BoolQueryBuilder? {
+    with(matchRequest) {
+      return dateOfBirth?.let {
+        QueryBuilders.boolQuery()
+            .mustWhenTrue({ activeSentence }, "currentDisposal", "1")
+            .mustMultiMatch(firstName, "firstName", "offenderAliases.firstName")
+            .mustWhenPresent("surname", surname)
+            .mustMatchOneOf("dateOfBirth", allLenientDateVariations(dateOfBirth))
+      }
+    }
+  }
+
+  private fun allLenientDateVariations(date:  LocalDate) : List<LocalDate> {
+    return swapMonthDay(date) + everyOtherValidMonth(date) + aroundDateInSameMonth(date)
+  }
+
+  private fun aroundDateInSameMonth(date: LocalDate) =
+      listOf(date.minusDays(1), date.minusDays(-1), date).filter { it.month == date.month }
+
+  private fun everyOtherValidMonth(date: LocalDate): List<LocalDate> =
+      (1..13).filterNot { date.monthValue == it }.mapNotNull { setMonthDay(date, it) }
```

Comment:
I'm sure there's a reason or month 13 - just wondering what it is?

Reply:
Ah good spot, I copied it from Java IntRange code which I think would end up with 1..12 - let me double check